xdsl/dialects/builtin.py

```diff
@@ -1405,6 +1405,18 @@ def get_shape(self) -> tuple[int, ...]:
     def get_element_type(self) -> AttributeCovT:
         return self.element_type
 
+    @classmethod
```

Comment:
Yeah the point is that we can't actually make them class methods IIRC, so we made them static methods. This is because there are things that classmethods let you write where the resulting constraint is inconsistent with the type signature, like `TensorType[A].constr()` which is actually a constraint for `TensorType[Attribute]` and will cause problems. Making the element_type non-optional may fix this issue.

Reply:
Oh right, I copied the `DenseArrayBase` one which is a `classmethod` (whoops)